lmfdb/genus2_curves/main.py

```diff
@@ -290,6 +289,45 @@ def genus2_lookup_equation(f):
             return r['label']
     return None
 
+
+def geom_inv_to_G2(inv):
+    def igusa_clebsch_to_G2(Ilist):
+        # first to Igusa
+        I2, I4, I6, I10 = Ilist
+        J2 = I2/8
+        J4 = (4*J2**2 - I4)/96
+        J6 = (8*J2**3 - 160*J2*J4 - I6)/576
+        J8 = (J2*J6 - J4*2)/4 # we won't use this one at all
+        J10 = I10/4096
+        return igusa_to_G2([J2, J4, J6, J8, J10])
+
+    def igusa_to_G2(Jlist):
+        monomials = [
```

Comment:
This code is pretty unreadable, and I don't see how it is really any better than a direct definition of the G2 invariants in terms of the Ji

Reply:
Maybe I should have been more clear, the monomials part is ok, but slicing the list based on certain terms being zero because they only contained numerators already checked to be zero makes my head hurt.

Reply:
Ok looks good now :)